### PR TITLE
[ETCM-460] Persist TnC acceptance

### PR DESCRIPTION
**Pull request by conqeror**

### CODE REVIEW

File: src/common/wallet-state.ts [hunk 1]
```diff
@@ -99,7 +85,25 @@ interface SendTransactionParams {
   nonce: number
 }
 
-export interface LoadedState {
+// States
+
+interface CommonState {
+  tncAccepted: boolean
+  setTncAccepted: (value: boolean) => void
```
Comment (qwhex) [inline src/common/wallet-state.ts:92]: Isn't it only relevant in NoWalletState?

Comment (conqeror) [reply src/common/wallet-state.ts:92]: Well...right now it is displayed only as a part of the wallet setup, but it doesn't mean that it does not make sense in other states. I can imagine that in the future when TnC are updated, we will want to display them to all users regardless of the state.

Comment (qwhex) [reply src/common/wallet-state.ts:92]: If the T&C changed, it doesn't matter if they accepted the previous one.

Comment (qwhex) [reply src/common/wallet-state.ts:92]: Hm, we still need a way to persist the acceptance of the new T&Cs in that case, so let's keep it this way.
We'll need a migration if it ever changes.

Comment (conqeror) [reply src/common/wallet-state.ts:92]: Yeah, that was exactly my thinking :)

**Review by qwhex — APPROVED** [review]

**Pull request merged**